browser/brave_app_controller_mac.mm

```diff
@@ -87,22 +148,70 @@ - (void)menuNeedsUpdate:(NSMenu*)menu {
   }
 }
 
+- (void)torMenuNeedsUpdate {
+  _torMenuItem.enabled = [self validateUserInterfaceItem:_torMenuItem];
+  _torMenuItem.hidden = !_torMenuItem.enabled;
+}
+
 - (BOOL)validateUserInterfaceItem:(id<NSValidatedUserInterfaceItem>)item {
   NSInteger tag = [item tag];
   if (tag == IDC_COPY_CLEAN_LINK) {
     return [self shouldShowCleanLinkItem];
   }
+
+  SEL action = [item action];
+  if (action == @selector(commandFromDock:) &&
+      tag == IDC_NEW_OFFTHERECORD_WINDOW_TOR) {
+    return !brave::IsTorDisabledForProfile([self lastProfileIfLoaded]) &&
```

Comment:
Could you please change the order, "canOpenNewBrowser" will check "lastProfileIfLoaded" before we have a chance to crash due to an invalid profile.

Reply:
Good call! Done.